wordpress-mobile/WordPress-iOS
# Issue/8277 site creation site details

Fixes #8277 

Needs Review: @nheagy 

**To test:**
Start the Site Creation process. Select a Theme. Verify the Site Details view appears.

**Behavior:**
- The Next button is disabled until a Site Title is entered.
- The Site Title is required; the Tagline is optional.
- Site Title field: the keyboard Next button goes to the Site Tagline field.
- Site Tagline field: the keyboard Done button emulates the Next button action.
- If the Next button is somehow enabled without a title, pressing the Next button will present a FancyAlertViewController error.
- Tapping the view will dismiss the keyboard.
- The text field clear button appears when editing.
- The view height adjusts with the keyboard, resulting in the Next button being repositioned with the keyboard.

Pressing Next with a valid Title will present a temporary confirmation alert:

<img width="272" alt="site_details_confirmation_alert" src="https://user-images.githubusercontent.com/1816888/33960905-765d20a2-e009-11e7-886e-454f6a32bd06.png">

**Implemented:**

![site_details](https://user-images.githubusercontent.com/1816888/34167368-d305eaa2-e49e-11e7-8274-1ecf2d9c8bec.png)

**Designed:**

<img width="250" alt="site_details_design" src="https://user-images.githubusercontent.com/1816888/33961157-3ecf29b8-e00a-11e7-8f6c-2a9b933c4bfe.png">

======================

![site_details](https://user-images.githubusercontent.com/1816888/34167724-c4b89de0-e49f-11e7-874c-c4ce541626f1.gif)
